File: app/src/main/java/org/dhis2/utils/Preconditions.java
package org.dhis2.utils;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public final class Preconditions {
    private Preconditions() {
        // no instances
    }

    @NonNull
    public static <T> T isNull(@Nullable T obj, @NonNull String message) {
        if (obj == null) {
            throw new IllegalArgumentException(message);
        }

        return obj;
    }

    public static boolean equals(@Nullable Object one, @Nullable Object two) {
        return one == two || one != null && one.equals(two); // NOPMD
    }
}

File: app/src/main/java/org/dhis2/utils/Preconditions.kt
package org.dhis2.utils

class Preconditions {

    companion object {

        @JvmStatic
        fun <T> isNull(obj: T?, message: String) =
            obj?.let { it } ?: throw IllegalArgumentException(message)

        @JvmStatic
        fun equals(one: Any?, two: Any?) = one == two || (one != null && one == two) // NOPMD
    }
}
